sentry-core/src/utils.rs

```diff
@@ -2,6 +2,121 @@
 
 use crate::protocol::DebugImage;
 
+/// Split the str into a typename and optional module prefix.
+///
+/// # Examples
+///
+/// ```
+/// use sentry_core::utils::parse_type_name;
+///
+/// let parsed = parse_type_name(std::any::type_name::<Vec<Option<usize>>>());
+/// assert_eq!(parsed, (Some("alloc::vec".into()), "Vec<core::option::Option<usize>>".into()));
+/// ```
+pub fn parse_type_name(mut type_name: &str) -> (Option<String>, String) {
```

Comment:
Can this return borrowed strings? 

A few thoughts on this:

- Do we need the generics? IIRC this was just used to get the error name, in which case I'd even prefer without generics. 
- It seems that we strip off the module path, but not module paths of the generics. This seems inconsistent.
- If we really need generics, do we need `dyn`?
- Ideally this would just slice out the relevant part(s), but not allocate. If we need to allocate we could do that in the caller.